src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -98,28 +204,67 @@ fn it_works() {
 Create a new file named `hello_fuchsia.rs` and fill out its contents with that
 code.
 
-### Create a package
+Using your freshly compiled `rustc`, you can compile a binary for Fuchsia using
+the following options:
+
+* `--target x86_64-fuchsia`/`--target aarch64-fuchsia`: Targets the Fuchsia
+  platform of your choice
+* `-Lnative ${SDK_PATH}/arch/${ARCH}/lib`: Link against Fuchsia libraries from
+  the SDK
+* `-Lnative ${SDK_PATH}/arch/${ARCH}/sysroot/lib`: Link against Fuchsia kernel
+  libraries from the SDK
+
+Putting it all together:
+
+```sh
+# Configure these for the Fuchsia target of your choice
+TARGET_ARCH="<x86_64-fuchsia|aarch64-fuchsia>"
+ARCH="<x64|aarch64>"
+
+rustc --target ${TARGET_ARCH} -Lnative=${SDK_PATH}/arch/${ARCH}/lib -Lnative=${SDK_PATH}/arch/${ARCH}/sysroot/lib -o hello_fuchsia hello_fuchsia.rs
+```
+
+Copy this binary into an intermediate directory, to be used in later steps:
+
+```sh
+mkdir fuchsia_rust_binary
+cp hello_fuchsia fuchsia_rust_binary/hello_fuchsia
+```
+
+**Current directory structure**
+```txt
+hello_fuchsia.rs
+hello_fuchsia              <-
+fuchsia_rust_binary/        | Copied binary
+┗━ hello_fuchsia           <-
+```
+
+## Creating a Fuchsia package
 
 On Fuchsia, a package is the unit of distribution for software. We'll need to
 create a new package directory where we will place files like our finished
-binary and any data it may need. The working directory will have this layout:
+binary and any data it may need.
 
+To start, make the `fuchsia_components/package`, `fuchsia_components/package/bin`, and `fuchsia_components/package/meta` directories:
+
+```sh
+mkdir -p fuchsia_components/package/bin
+mkdir fuchsia_components/package/meta
+```
+
+**Current directory structure**
 ```txt
-hello_fuchsia.rs
-hello_fuchsia.cml
-package
-┣━ bin
-┃  ┗━ hello_fuchsia
-┣━ meta
-┃  ┣━ package
-┃  ┗━ hello_fuchsia.cm
-┗━ hello_fuchsia.manifest
+fuchsia_rust_binary/
+┗━ ...
+fuchsia_components/
```

Comment:
This directory tree is having a rough day

Reply:
Oh goodness, that extra "┃ " is not very helpful :joy_cat: thanks for the spot!